attrib/main.c

```diff
@@ -0,0 +1,162 @@
+/*
+	main.c (20.01.11)
+	Prints or changes exFAT file attributes
+
+	Free exFAT implementation.
+	Copyright (C) 2020       Endless OS Foundation LLC
+
+	This program is free software; you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 2 of the License, or
+	(at your option) any later version.
+
+	This program is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License along
+	with this program; if not, write to the Free Software Foundation, Inc.,
+	51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+#include <exfat.h>
+#include <stdio.h>
+#include <string.h>
+#include <unistd.h>
+
+static void usage(const char* prog)
+{
+	fprintf(stderr,
+		"Display current attributes:\n"
+		"  %1$s <device> <file>\n"
+		"\n"
+		"Set attributes:\n"
+		"  %1$s [FLAGS] <device> <file>\n"
+		"\n"
+		"Flags:\n"
+		"  -r    Set read-only flag\n"
+		"  -R    Clear read-only flag\n"
+		"  -i    Set hidden flag\n"
+		"  -I    Clear hidden flag\n"
+		"  -s    Set system flag\n"
+		"  -S    Clear system flag\n"
+		"\n"
+		"  -h    Display this help message\n"
+		"  -V    Display version information\n",
+		prog);
+	exit(1);
+}
+
+static void print_attribute(uint16_t attribs, uint16_t attrib, const char *label)
+{
+	printf("%9s: %s\n", label, (attribs & attrib) ? "yes" : "no");
+}
+
+int main(int argc, char* argv[])
+{
+	int opt;
+	int ret;
+	const char* spec = NULL;
+	const char* options = "";
+	const char* file_path = NULL;
+	struct exfat ef;
+	struct exfat_node* node;
+	uint16_t add_flags = 0;
+	uint16_t clear_flags = 0;
+
+	while ((opt = getopt(argc, argv, "rRiIsShV")) != -1)
+	{
+		switch (opt)
+		{
+		case 'V':
+			printf("exfatattrib %s\n", VERSION);
+			puts("Copyright (C) 2011-2018  Andrew Nayenko");
+			puts("Copyright (C) 2020       Endless OS Foundation LLC");
+			return 0;
+		case 'r':
+			add_flags |= EXFAT_ATTRIB_RO;
+			break;
+		case 'R':
+			clear_flags |= EXFAT_ATTRIB_RO;
+			break;
+		/* "-h[elp]" is taken; i is the second letter of "hidden" and
+		 * its synonym "invisible"
+		 */
+		case 'i':
+			add_flags |= EXFAT_ATTRIB_HIDDEN;
+			break;
+		case 'I':
+			clear_flags |= EXFAT_ATTRIB_HIDDEN;
+			break;
+		case 's':
+			add_flags |= EXFAT_ATTRIB_SYSTEM;
+			break;
+		case 'S':
+			clear_flags |= EXFAT_ATTRIB_SYSTEM;
+			break;
+		default:
+			usage(argv[0]);
+		}
+	}
+
+	if ((add_flags & clear_flags) != 0)
+	{
+		fprintf(stderr, "Can't set and clear the same flag\n");
```

Comment:
Please use `exfat_error` to print errors. They echo messages to syslog when an utility is detached from tty.